### DISCUSSION TITLE
Linux networking: bidirectional host-to-guest communication

### DISCUSSION THREAD

**Discussion by hadjiprocopis** [joncampbell123/dosbox-x#4303] [Q&A]
Hi all,
I want to share my experience on how to achieve bidirectional communication between a Linux box and a dosbox-x hosted there in.

In general, I have followed the tutorial here: https://dosbox-x.com/wiki/Guide%3ASetting-up-networking-in-DOSBox%E2%80%90X#_mtcp_example

I have used the slirp backend with this configuration with one important addition, the ```tcp_port_forwards = 6666``` which essentially allows me to "access" port 6666 in dosbox-x:

```
[ne2000]
ne2000=true
nicirq=10
backend=slirp
[ethernet, slirp]
mtu=1500
mru=1500
# I want to talk to dosbox-x / port 6666 from linux host:
tcp_port_forwards = 6666 

```
On running the dosbox-x I do:
```
MOUNT C mtcp
C:
SET MTCPCFG=C:\SAMPLES\SAMPLE.CFG
NE2000 0x60 10 0x300
DHCP
```
DHCP talks back to me and gives me:
```
IPADDR 10.0.2.15
NETMASK 255.255.255.0
GATEWAY 10.0.2.2
NAMESERVER 10.0.2.3
```

From dosbox-x I can contact a simple diy echo server running on my linux box (hosting dosbox-x) and listening on port 3333, with ```telnet.exe 10.0.2.2 3333```. So, I can confirm bidirectional communication when the server is on linux box.

Then I run on the dosbox-x a simple echo server based on mTCP (which reads the same configuration as telnet.exe, i.e. with IPADDR etc.) and which listens to port 6666.

What I did not find (I think!) in the tutorial is that I can access the dosbox-x echo server from my linux box (hosting the dosbox-x) using ```telnet 127.0.0.1 6666```. This is possible thanks to the line under ```[ethernet, slirp]``` : ```tcp_port_forwards = 6666```. Also note also that from linux's point of view the dosbox-x is at ```127.0.0.1``` and from dosbox-x's point of view the linux host is at ```10.0.2.2```

**Comment by rderooy** [comment]
Thank you.

I should find some time to add an example to the wiki on setting up slirp with port forwarding.

**Comment by sys49152** [comment]
Hi all,
any news on that matter?
